### PR TITLE
Fix/vps snapshot option

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

Fix the api call 

### Needs 

- [x] Waiting for 2API 

SCFRCA-2117

### CODE REVIEW

File: client/app/vps/import/vps.service.js [hunk 1]
```diff
@@ -674,34 +674,35 @@ angular.module("managerApp").service("VpsService", [
             });
         };
 
-        // HOT FIX remove this fukin shit
-        this.getOptionDetails2 = function (option, vps, duration) {
+        // HOT FIX
```
Comment (FredericEspiau) [inline client/app/vps/import/vps.service.js]: This doesn't look like a hotfix anymore, is it ? 😄

File: client/app/vps/import/vps.service.js [hunk 2]
```diff
@@ -674,34 +674,35 @@ angular.module("managerApp").service("VpsService", [
...
+        this.getSnapshotDetails = function (vps, duration) {
+            const version = vps.version.replace(/_/g, "").toLowerCase();
```
Comment (FredericEspiau) [inline client/app/vps/import/vps.service.js]: Could we have an example in comments please ?

`// from "_2018_V_1" to "2018v1"`

File: client/app/vps/import/vps.service.js [hunk 3]
```diff
@@ -674,34 +674,35 @@ angular.module("managerApp").service("VpsService", [
...
-        // HOT FIX remove this fukin shit
+        // HOT FIX
         this.getOptionSnapshot = function (vps) {
-            return $http.get(["apiv6/order/vps/", vps.name, "/snapshot"].join(""));
+            return $http.get(`/order/vps/${vps.name}/snapshot`);
         };
 
         this.getOptionSnapshotFormated = function (serviceName, vps) {
-            if (vps.version === "_2015_V_1" && vps.offerType === "SSD") {
+            if (_.includes(["_2015_V_1", "_2018_V_1"], vps.version) && vps.offerType === "SSD") {
```
Comment (FredericEspiau) [inline client/app/vps/import/vps.service.js]: Will we have to update again for `_2019_V_1` ?

File: client/app/vps/import/vps.service.js [hunk 4]
```diff
@@ -674,34 +674,35 @@ angular.module("managerApp").service("VpsService", [
...
-            } else {
-                return this.getOptionDetails(serviceName, "snapshot").then(data => data.results[0]);
+                    .then(snapshotOption => this.getSnapshotDetails(vps, snapshotOption.data[0]))
+                    .then(([price, snapshotOrderDetails, duration]) => ({
+                        unitaryPrice: price.data.text,
+                        withoutTax: snapshotOrderDetails.data.prices.withoutTax.text,
+                        withTax: snapshotOrderDetails.data.prices.withTax.text,
+                        duration: { duration }
+                    })
+                    );
```
Comment (FredericEspiau) [inline client/app/vps/import/vps.service.js]: Indentation seems a bit off 😃

File: client/app/vps/import/vps.service.js [hunk 5]
```diff
@@ -674,34 +674,35 @@ angular.module("managerApp").service("VpsService", [
...
             }
+            return this.getOptionDetails(serviceName, "snapshot").then(data => data.results[0]);
+
         };
 
         // HOT FIX remove this fukin shit
```
Comment (FredericEspiau) [inline client/app/vps/import/vps.service.js]: Please finish the job with the horrible comments 😃

File: client/app/vps/import/vps.service.js [hunk 6]
```diff
@@ -668,43 +668,18 @@ angular.module("managerApp").service("VpsService", [
...
-                                duration: {duration: d.data[0]}
-                            }
-                        });
-            } else {
-                return this.getOptionDetails(serviceName, "snapshot").then(data => data.results[0]);
-            }
         };
 
-        // HOT FIX remove this fukin shit
+        // HOT FIX
```
Comment (FredericEspiau) [inline client/app/vps/import/vps.service.js]: Not a hot fix anymore, is it ?

File: client/app/vps/import/vps.service.js [hunk 7]
```diff
@@ -668,43 +668,18 @@ angular.module("managerApp").service("VpsService", [
...
-            ]);
+            });
         };
 
-        // HOT FIX remove this fukin shit
-        this.getOptionSnapshot = function (vps) {
-            return $http.get(["apiv6/order/vps/", vps.name, "/snapshot"].join(""));
-        };
+        this.getOptionSnapshotFormated = function (serviceName) {
+            return this.getOptionDetails(serviceName, "snapshot").then(data => data.results[0]);
```
Comment (JeremyDec) [inline client/app/vps/import/vps.service.js]: `_.first(data.results)` ?

**Pull request merged**